VideoAPI/Testing.Common/Helper/ZAP.cs

```diff
@@ -0,0 +1,168 @@
+using Microsoft.Extensions.Configuration;
+using OWASPZAPDotNetAPI;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Threading;
+using Testing.Common.Configuration;
+
+namespace Testing.Common.Helper
+{
+    public static class ZAP
+    {        
+        private static readonly ClientApi _api = new ClientApi(ZapConfiguration.ApiAddress, ZapConfiguration.ApiPort, ZapConfiguration.Apikey);
+        private static IApiResponse _apiResponse;
+
+        private static ZAPConfiguration ZapConfiguration
+        {
+            get
+            {
+                return new ConfigurationBuilder()
+                    .AddJsonFile("appsettings.json")
+                    .AddUserSecrets("9AECE566-336D-4D16-88FA-7A76C27321CD")
+                    .Build()
+                    .GetSection("ZAPConfiguration")
+                    .Get<ZAPConfiguration>();
+            }
+        }
+
+        public static IWebProxy WebProxy
+        {
+            get
+            {
+                return ZapConfiguration.RunZap ? new WebProxy($"http://{ZapConfiguration.ApiAddress}:{ZapConfiguration.ApiPort}", false) : null;
+            } 
+        }
+         
+
+        public static void StartZAPDaemon()
+        {
+            if (!ZapConfiguration.RunZap) return;
+
+            var zapProcessStartInfo = new ProcessStartInfo
+            {
+                FileName = @$"{ZapConfiguration.WorkingDirectory}\ZAP.exe",
+                WorkingDirectory = ZapConfiguration.WorkingDirectory,
+                Arguments = $"-daemon -host {ZapConfiguration.ApiAddress} -port {ZapConfiguration.ApiPort}"
+            };
+             
+            Process.Start(zapProcessStartInfo);
+             
+            CheckIfZAPHasStartedByPollingTheAPI(1);
+        } 
+
+        public static void CheckIfZAPHasStartedByPollingTheAPI(int minutesToWait)
+        { 
+            var watch = new Stopwatch();
+            watch.Start();
+            var millisecondsToWait = ZapConfiguration.MinutesToWait * 60 * 1000;
+            var zapUrlToDownload = $"http://{ZapConfiguration.ApiAddress}:{ZapConfiguration.ApiPort}";
+
+            while (millisecondsToWait > watch.ElapsedMilliseconds)
```

Comment:
This will run Infinitely if the site is down. Maybe try to use Polly to do reties.. ?

Reply:
MilliSecondsToWait verifies the duration until which the loop is run, which is at present set to 1minute and ZAP kicks of well within the time frame.